### PR TITLE
chore: [APPAI-1850] Migrated deployment from saas to app-interface

### PR DESCRIPTION
**Pull request by dgpatelgit**
Migrate deployment of pypi insights service from saas to app-interface.

Jira ticket :: https://issues.redhat.com/browse/APPAI-1850

### CODE REVIEW

**Comment by arajkumar** [comment]
@dgpatelgit Since this doesn't have UBI migration changes, it should have one build script for rhel as well.

**Comment by sawood14012** [comment]
@dgpatelgit how about migrating it to UBI ?

**Review by sawood14012 — CHANGES_REQUESTED** [review]
if not using ubi i suggest you create the image mirror MR first on app-sre . and add the rhel build deploy file as arun said

**Comment by dgpatelgit** [comment]
> @dgpatelgit how about migrating it to UBI ?

Lets migrate to UBI image, I will make changes.

File: Dockerfile [hunk 1]
```diff
@@ -1,23 +1,24 @@
...
+ENV LANG=en_US.UTF-8 PYTHONDONTWRITEBYTECODE=1
 
-RUN pip3 install --upgrade pip
+RUN pip3 install --upgrade --no-cache-dir pip
 
-RUN pip3 install git+https://github.com/fabric8-analytics/fabric8-analytics-rudra#egg=rudra
+COPY ./requirements.txt /opt/app-root
 
-RUN pip3 install -r requirements.txt
+RUN pip3 install --no-cache-dir git+https://github.com/fabric8-analytics/fabric8-analytics-rudra#egg=rudra
```
Comment (arajkumar) [inline Dockerfile:17]: Can this be moved into requirements.in?

File: Dockerfile [hunk 2]
```diff
@@ -1,23 +1,24 @@
...
+RUN pip3 install --no-cache-dir -r /opt/app-root/requirements.txt
+RUN pip3 install --no-cache-dir Cython==0.29.1 && pip3 install --no-cache-dir hpfrec==0.2.2.9
```
Comment (arajkumar) [inline Dockerfile:19]: Ditto

**Review by arajkumar — CHANGES_REQUESTED** [review]
Since you have removed all supporting files related to CICO builds, make sure to remove the GH webhook before merging this PR.

File: Dockerfile (hunk 2 above)
Comment (dgpatelgit) [reply Dockerfile:17]: @arajkumar we cannot move cython and hpfrec into requirements.in as it generates dependency conflicts with common module between them and rudra for various modules like `numpy`, `jinja2`, `pygments` etc. We cannot resolve them without altering rudra dependencies.

Comment (arajkumar) [reply Dockerfile:17]: I see, then rudhra should be cleaned up(majorly setup.py should refer requirements.in). It shall go as a separate PR.

File: qa/runtest.sh [hunk 3]
```diff
@@ -1,31 +1,75 @@
...
+echo "*****************************************"
+radon mi -s -i venv .
+
+echo "*****************************************"
+echo "*** Unit tests ***"
+echo "*****************************************"
+PYTHONDONTWRITEBYTECODE=1 python3 "$(which pytest)" --cov=src/ --cov-report=xml --cov-fail-under=$COVERAGE_THRESHOLD -vv tests/unit_tests/
+printf "%stests passed%s\n\n" "${GREEN}" "${NORMAL}"
+
+codecov --token=74b5a608-da00-4b26-aec8-8f7f47489f86
```
Comment (arajkumar) [inline qa/runtest.sh:73]: This is not required, GH codecov action takes care of this

Comment (dgpatelgit) [reply qa/runtest.sh:73]: @arajkumar GH action triggered this script for unit test cases, this line is very much required to trigger unit test cases.

Comment (arajkumar) [reply qa/runtest.sh:73]: I was mentioning about the following line

```
codecov --token=74b5a608-da00-4b26-aec8-8f7f47489f86
```

File: Dockerfile (hunk 2 above)
Comment (dgpatelgit) [reply Dockerfile:17]: Agree, we need to take-up rudra clean up as separate task.

**Review by arajkumar — APPROVED** [review]
lgtm

**Review by sawood14012 — APPROVED** [review]

**Pull request merged**